src/krate/mod.rs

```diff
@@ -602,18 +602,19 @@ pub fn index(req: &mut Request) -> CargoResult<Response> {
 
     if let Some(q_string) = params.get("q") {
         let sort = params.get("sort").map(|s| &**s).unwrap_or("relevance");
+        let canon_q_string = canon_crate_name(q_string);
         let q = plainto_tsquery(q_string);
         query = query.filter(
             q.matches(crates::textsearchable_index_col)
-                .or(crates::name.eq(q_string)),
+                .or(canon_crate_name(crates::name).eq(canon_q_string)),
```

Comment:
This is getting repeated quite a bit. Can we pull it out into a function? (Worth extracting from `Crate::by_name` as well).

Something like this should work:

```rust
use diesel::expression::helper_types::Eq;

fn name_canonically_equals(s: &str) -> Eq<canon_crate_name<crates::name>, canon_crate_name<&str>> {
    canon_crate_name(crates::name).eq(canon_crate_name(s))
}
```

Reply:
I honestly didn't know that was possible. Thanks! I'll work on it soon.